### PR TITLE
Feat: loading screen - avatar head look at mouse cursor (IK)

### PR DESCRIPTION
**Pull request by popuz**
# Pull Request Description

### Prerequisites
- [ ] List any required setup steps
- [ ] Include environment/configuration requirements

### Test Steps
1. First step
2. Second step
3. Expected result after step 2
4. ...

### Additional Testing Notes
- Note any edge cases to verify
- Mention specific areas that need careful testing
- List known limitations or potential issues

## Quality Checklist
- [ ] Changes have been tested locally
- [ ] Documentation has been updated (if required)
- [ ] Performance impact has been considered
- [ ] For SDK features: Test scene is included

## Code Review Reference
Please review our [Code Review Standards](https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md) before submitting.

### CODE REVIEW

**Review by NickKhalow — CHANGES_REQUESTED** [review]
Looks good! Please address the comments

**Review by anicalbano — APPROVED** [review]
✔️ Reviewed by QA and working as expected.
✔️ All edge cases where checked and no further issues were found
✔️ The avatar behavior remains normal in all other instances

https://github.com/user-attachments/assets/94163eb1-e5d0-4e56-a6ed-3e4940537373


https://github.com/user-attachments/assets/324b9c4a-0878-4291-bfa4-8fcac75844c1

File: Explorer/Assets/DCL/Character/CharacterMotion/Systems/HeadIKSystem.cs [hunk 1]
```diff
@@ -33,9 +36,15 @@ public partial class HeadIKSystem : BaseUnityLoopSystem
         private readonly ElementBinding<float> horizontalReset;
         private readonly ElementBinding<float> speed;
         private SingleInstanceEntity settingsEntity;
+        private readonly ICharacterControllerSettings settings;
+        private readonly DCLInput dclInput;
+        private readonly Vector3[] previewImageCorners = new Vector3[4];
```
[the remark this answers is not in the document]
Comment (NickKhalow) [reply Explorer/Assets/DCL/Character/CharacterMotion/Systems/HeadIKSystem.cs:41]: GetWorldCorners doesn't return an array, it populates an existing one

Therefore I see it could be used as `previewComponent.RenderImageRect.GetWorldCorners(corners.Raw);`

I would avoid dealing with explicit indexes because it's error prone and to deal with it you have to keep in mind order of corners and ensure that the size of the array is correct

Moreover there is a bad thing in their API that it's silent fail and won't explicitly tell you that something went wrong in their contract
```
    public void GetWorldCorners(Vector3[] fourCornersArray)
    {
      if (fourCornersArray == null || fourCornersArray.Length < 4)
      {
        Debug.LogError((object) "Calling GetWorldCorners with an array that is null or has less than 4 elements.");
      }
```

WDYT?

[the remark this answers is not in the document]
Comment (popuz) [reply Explorer/Assets/DCL/Character/CharacterMotion/Systems/HeadIKSystem.cs:41]: agree, now it looks much more readable and even shorter

**Review by NickKhalow — APPROVED** [review]
Great Work 🚀 
Approved
